src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PartiallyDeterminedHaplotypeComputationEngine.java

```diff
@@ -382,39 +312,76 @@ private static List<List<Event>> smithWatermanRealignPairsOfVariantsForEquivalen
     }
 
     /**
-     * Partition events into clusters that must be considered together, either because they overlap or because they belong to the
-     * same mutually exclusive pair or trio.  To find this clustering we calculate the connected components of an undirected graph
-     * with an edge connecting events that overlap or are mutually excluded.
+     * Partition events into the largest possible clusters such that events in distinct clusters are mutually compatible
```

Comment:
I'm not sure this description covers that in the average case you are ending up with a bunch of singleton event groups... you could say the maximum nubmer of clusters such that.... but also I would maybe spell out the consequences of this? Maybe even with an example with an external SNP - A and a Del + SNP that overlap showing that this would give you two event groups?

Reply:
I would make a note for posterity here or elsewhere that this is a significant divergence from the dagen code.

Reply:
I added a note about the common case where all events are compatible and a single EventGroup results.

As far as I understand the EventGroups that this yields are identical to DRAGEN's, but I added a note that the algorithm is significantly different.